File: Frontend/src/app/core/services/cinema.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CinemaPayload, CinemaSummary } from '../models/cinema.model';

@Injectable({ providedIn: 'root' })
export class CinemaService {
  private readonly baseUrl = '/api/cinemas';

  constructor(private http: HttpClient) {}

  getCinemas(): Observable<CinemaSummary[]> {
    return this.http.get<CinemaSummary[]>(this.baseUrl);
  }

  createCinema(payload: CinemaPayload): Observable<CinemaSummary> {
    return this.http.post<CinemaSummary>(this.baseUrl, payload);
  }

  updateCinema(id: number, payload: CinemaPayload): Observable<CinemaSummary> {
    return this.http.put<CinemaSummary>(`${this.baseUrl}/${id}`, payload);
  }

  deleteCinema(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: cinema-ticket-java/src/main/java/com/cinema/ticket/system/service/CinemaService.java
package com.cinema.ticket.system.service;

import com.cinema.ticket.system.dto.CinemaDTO;
import com.cinema.ticket.system.model.Cinema;

import java.util.List;

public interface CinemaService {
    
    List<CinemaDTO> getAllCinemas();
    
    Cinema getCinemaById(Long id);

    CinemaDTO createCinema(CinemaDTO cinemaDTO);

    CinemaDTO updateCinema(Long id, CinemaDTO cinemaDTO);

    void deleteCinema(Long id);
}
